internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -653,30 +653,61 @@ func NewJfrDatasourceContainer(cr *operatorv1beta1.Cryostat, imageTag string) co
 	}
 }
 
-func NewExporterService(cr *operatorv1beta1.Cryostat) *corev1.Service {
+func NewCoreService(cr *operatorv1beta1.Cryostat) *corev1.Service {
+	// Initialize with defaults
+	svcType := corev1.ServiceTypeClusterIP
+	httpPort := int32(8181)
+	jmxPort := int32(9091)
+	labels := map[string]string{}
+	annotations := map[string]string{}
+
+	// Override with any user-specified properties
+	if cr.Spec.ServiceOptions != nil && cr.Spec.ServiceOptions.CoreConfig != nil {
+		config := cr.Spec.ServiceOptions.CoreConfig
+		if config != nil {
```

Comment:
Can it be `nil` here? Seems like the RHS of the first `if` two lines up should prevent that

Reply:
Right, I've removed this redundant check.